layouts/ucf-resource-link-classic.php

```diff
@@ -0,0 +1,303 @@
+
+		// Set up a post query
+		$query_args = array(
+			'numberposts' => -1,
+			'post_type'   => $args['post_type_name'],
+			'tax_query'   => array(
+				array(
+					'taxonomy' => $args['taxonomy'],
+					'field'    => 'id',
+					'terms'    => '',
+				)
+			),
+			'orderby'     => $args['order_by'],
+			'order'       => $args['order'],
+		);
+
+		// Handle meta key and value query
+		if ( $args['meta_key'] && $args['meta_value'] ) {
+			$query_args['meta_key'] = $args['meta_key'];
+			$query_args['meta_value'] = $args['meta_value'];
+		}
+		// Split up this post type's posts by term
+		$by_term = array();
+		foreach ( get_terms( $args['taxonomy'] ) as $term ) { // get_terms defaults to an orderby=name, order=asc value
+			$query_args['tax_query'][0]['terms'] = $term->term_id;
+			$posts = get_posts( $query_args );
+			if ( count( $posts ) == 0 && $args['show_empty_sections'] ) {
+				$by_term[$term->name] = array();
+			} else {
+				$by_term[$term->name] = $posts;
+			}
+		}
+		// Add uncategorized items to posts by term if parameter is set.
+		if ( $args['show_uncategorized'] ) {
+			$terms = get_terms( $args['taxonomy'], array( 'fields' => 'ids', 'hide_empty' => false ) );
+			$query_args['tax_query'][0]['terms'] = $terms;
+			$query_args['tax_query'][0]['operator'] = 'NOT IN';
+			$uncat_posts = get_posts( $query_args );
+			if ( count( $uncat_posts == 0 ) && $args['show_empty_sections'] ) {
+				$by_term[$args['uncategorized_term_name']] = array();
+			} else {
+				$by_term[$args['uncategorized_term_name']] = $uncat_posts;
+			}
+		}
+		// Split up this post type's posts by the first alpha character
+		$query_args['orderby'] = 'title';
+		$query_args['order'] = 'ASC';
+		$query_args['tax_query'] = '';
+		$by_alpha_posts = get_posts( $query_args );
+		foreach( $by_alpha_posts as $post ) {
+			if ( preg_match( '/([a-zA-Z])/', $post->post_title, $matches ) == 1 ) {
+				$by_alpha[strtoupper($matches[1])][] = $post;
+			} else {
+				$by_alpha[$args['non_alpha_section_name']][] = $post;
+			}
+		}
+		if( $args['show_empty_sections'] ) {
+			foreach( range( 'a', 'z' ) as $letter ) {
+				if ( !isset( $by_alpha[strtoupper( $letter )] ) ) {
+					$by_alpha[strtoupper( $letter )] = array();
+				}
+			}
+		}
+		ksort( $by_alpha );
+		$sections = array(
+			'resource-search-term'  => $by_term,
+			'resource-search-alpha' => $by_alpha,
+		);
+		ob_start();
+	?>
+		<div class="resource-search">
+			<div class="resource-search-header">
+				<form class="resource-search-form" action="." method="get">
+					<label class="resource-search-label" for="resource-search-input">
+						<h2>
+							<?php echo $args['default_search_label']; ?>
+						</h2>
+					</label>
+					<input type="text" id="resource-search-input" name="resource-search-input" class="resource-search-input" placeholder="<?php echo $args['default_search_text']; ?>">
+				</form>
+			</div>
+			<div class="resource-search-results"></div>
+			<?php if ( $args['show_sorting'] ) { ?>
```

Comment:
Style guide: We should use the alternate syntax for large blocks like this:

`<?php if ( $args['show_sorting'] ) : ?>`